### ISSUE TITLE
AddMany-Mdbc

### ISSUE THREAD

**Issue by jb-home** [nightroman/Mdbc#77]
Hi,

would it be possible to add the add many functionality?
Currently it seems that your example:
@{_id = 1; value = 42}, @{_id = 2; value = 3.14} | Add-MdbcData

Would do two requests and not a single add many request.

**Comment by nightroman** [comment]
Whatever is supported by the C# driver is in theory possible. In practice, several ways to implement this, pros and cons, and etc. should be evaluated and some time found. I have not planned this feature so far.

**Comment by nightroman** [comment]
I am closing this. This was not planned and is still not planned. There is no convincing evidences that this feature brings more good (performance, something else?) than bad (increased complexity, both internal and UX, etc.).

**Comment by Nicarex** [comment]
Hello, Roman! 
 
I've encountered a situation where a small dataset of 100,000 objects has to be inserted into a MongoDB collection. Iterating through each object using ForEach {Add-MdbcData -InputObject $Add} takes about 10 minutes, which is very slow... 
 
Using my limited experience in Rust, I wrote a simple program that takes data from a JSON file and inserts it into Mongo using InsertMany (Rust Driver). The same dataset of 100,000 objects (after converting it to JSON beforehand) was inserted in just 5 seconds. The difference is crucial! However, I still couldn't solve the issue with the object types; 'DateTime' gets persistently lost during the JSON conversion. 
 
I would really like to see the implementation of the InsertMany function in the Mdbc module 🙂
